### PR TITLE
Pragmas, compound lits, etc.

### PR DESCRIPTION
**Pull request by chathhorn**
@dwightguth

### CODE REVIEW

File: semantics/language/execution/compound-literal.k [hunk 1]
```diff
@@ -57,10 +57,23 @@ module C-COMPOUND-LITERAL
                ~> X
           ...</k>
           <compound-lit-map> M:Map => M[X <- Base] </compound-lit-map>
+          <local-addresses>... .Set => SetItem(Base) </local-addresses>
```
Comment (dwightguth) [inline semantics/language/execution/compound-literal.k]: can you explain to me what is going on with this change here?

File: semantics/language/translation/compound-literal.k [hunk 2]
```diff
@@ -58,15 +58,13 @@ module C-COMPOUND-LITERAL
...
-               ~> Init
+               ~> allowInit(Init)
                ~> X
           ...</k>
           <curr-tu> Tu:String </curr-tu>
           <tu-id> Tu </tu-id>
           <next-static-loc> Base:SymBase => linc(Base) </next-static-loc>
-          <initializing> I:Int </initializing>
-          requires I >Int 0
+          <block-history> .List </block-history>
```
Comment (dwightguth) [inline semantics/language/translation/compound-literal.k]: and here?

File: semantics/language/execution/compound-literal.k (hunk 1 above)
Comment (chathhorn) [inline semantics/language/execution/compound-literal.k]: This adds the newly allocated address to the set of local addresses so it'll be cleaned up/invalidated when the compound literal object goes out of scope. Compound literals at block scope are supposed to have a lifetime of the block.

File: semantics/language/translation/compound-literal.k (hunk 2 above)
Comment (chathhorn) [inline semantics/language/translation/compound-literal.k]: This prevents block-scope compound literals from being allocated at translation-time. Global-scope literals can be allocated, as they're supposed to have static storage class and a lifetime for the whole program.

**Comment by dwightguth** [comment]
Okay. If you fix the conflicts, you can merge this.

**Pull request merged**